tests/unit/extract/test_group_holdings.py

```diff
@@ -0,0 +1,324 @@
+import unittest
+from lusidtools import logger
+from datetime import datetime
+from parameterized import parameterized
+import pytz
+
+from lusidtools.extract.group_holdings import _join_holdings
+from lusid.models import VersionedResourceListOfPortfolioHolding, PortfolioHolding, Link, Version, ModelProperty, PropertyValue, CurrencyAndAmount
+
+now = datetime.now(pytz.UTC)
+
+class CocoonExtractGroupHoldingsTests(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.logger = logger.LusidLogger("debug")
+
+
+
+    class MockVersionedResourceListOfPortfolioHolding(VersionedResourceListOfPortfolioHolding):
+
+        def __init__(self, portfolio_holdings):
+            super().__init__(
+                values=portfolio_holdings,
+                version=Version(
+                    as_at_date=now,
+                    effective_from=now
+                ),
+                links=[Link(
+                    relation="MOCK",
+                    href="MOCK",
+                    method="MOCK"
+                )]
+            )
+
+    @parameterized.expand([
```

Comment:
its quite hard to read these tests give the size of the inputs, given the single function call inside the test, I wonder if its better to write them as separate explicitly tests?

Reply:
Found a much nicer solution, thanks!